### ISSUE TITLE
Support yarn pnp vitest installs

### ISSUE THREAD

**Issue by grapoza** [vitest-dev/vscode#58]
The resolution of the vitest version is dependent on the `node_modules` folder. However, with [Yarn pnp](https://yarnpkg.com/features/pnp) installs, there is no `node_modules` and so the [detection of the vitest version fails](https://github.com/vitest-dev/vscode/blob/main/src/pure/utils.ts#L101) with `Cannot get vitest version from "undefined"`.

It would be great to have this detection algorithm take Yarn pnp installs into account. There are two ways I could see this happening.

1) Look for the pnp package itself, which in my case is (relative to the folder root): `.yarn/cache/vitest-npm-0.18.0-a8155f7bdc-20e76b0de9.zip`
2) Look for the entry in the .pnp.cjs file, which in my case is in the folder root (pretty sure that's always the case?)
  ```
  ...
      ["vitest", [\
        ["npm:0.18.0", {\
          "packageLocation": "./.yarn/cache/vitest-npm-0.18.0-a8155f7bdc-20e76b0de9.zip/node_modules/vitest/",\
          "packageDependencies": [\
            ["vitest", "npm:0.18.0"]\
          ],\
          "linkType": "SOFT"\
        }],\
  ...
  ```

I'd be glad to look into this more and see if I can get a PR working, but it would have to be at a later date since I'm well past the [Ballmer Peak](https://xkcd.com/323/) tonight.

**Comment by grapoza** [comment]
I finally got a chance to run through this. While `getVitestCommand` does fail to resolve a command because there isn't a `node_modules` folder anywhere along the project root's path, the code is attempting to use an `npx vitest -v` command as a fallback. On my system this is failing in the [`spawn` command](https://github.com/vitest-dev/vscode/blob/50fb162f447493b32dccabb852c3757a3664fb5c/src/pure/utils.ts#L83) because it can't find npx (Windows 10, so if I change it to use `npx.cmd` it finds it). If I run the `spawn` with `shell: isWindows` it works since the shell can resolve npx. It looks like you're already using this approach in [`execWithLog`](https://github.com/vitest-dev/vscode/blob/50fb162f447493b32dccabb852c3757a3664fb5c/src/pure/utils.ts#L149).

If I run a test after that then I can see the test run correctly in the output but I get a `Failed to get any result` error. I would guess this is also related to my environment, but I'll look into it more. Given what I've found so far, it's likely this issue can be closed or renamed to better reflect any changes found that are worth going into the codebase since it's doubtful pnp packages are causing whatever this other failure is either.

```
Failed to get any result
( Vitest should be configured to be able to run from project root )

Error when running
yarn test D:/Code/JavaScript/vue-tree/tests/unit/TreeView.spec.js -t "TreeView.vue when on an element without an ID should have an autogenerated uniqueId prefixed with grt-"

cwd: D:/Code/JavaScript/vue-tree
node: v16.14.0
env.PATH: undefined

[Error 8:42:59 PM] 
 RUN  v0.18.0 D:/Code/JavaScript/vue-tree
      API started at http://localhost:50610

 ✓ tests/unit/TreeView.spec.js  (20 tests | 19 skipped) 21ms

Test Files  1 passed (1)
     Tests  1 passed | 19 skipped (20)
      Time  2.76s (in thread 21ms, 13150.92%)
```
